password-generator-manager/src/utils.js

```diff
@@ -43,17 +44,17 @@ export function generateRandomPassword(length) {
     specialCharacters[Math.floor(Math.random() * specialCharacters.length)];
   password += randomCharacter;
 
+  // Remove whitespace, if any
+  password = password.replace(/ +/g, "");
+
   // Generate the remaining characters
   const remainingLength = length - 3;
-  const characters = words.join("") + randomNumber + specialCharacters.join("");
+  const characters = alphabet + randomNumber + specialCharacters.join("");
```

Comment:
Hi @utk09!
 I'm not sure I understand,  but I can explain my thought process a bit more!

Some of elements in the **words** variable contains spaces, such as **'American Bobtail'**. If randomWord was set to **'American Bobtail'**, then the password would have a space. Since we don't want spaces in the password, line 
`password = password.replace(/ +/g, "");` removes them.

But there are still spaces in the words variable. Even if we join them using this:
` words.join("")`
the spaces inside the elements would like exist I think (eg. AbyssinianAmerican BobtailAmerican Curl)
So we could do something like:
'words.join("").replace(/ +/g, "");`

But like you said this calls the replace() method again and it not very readable. I thought the alphabet variable was more readable, but I'm open to any suggestions!